### PR TITLE
config: make EnableSlowLog atomic

### LINKED ISSUE #30345: data race in the GlobalConfig
## Bug Report

Please answer these questions before submitting your issue. Thanks!

```
 WARNING: DATA RACE
Write at 0x00c0095f6840 by goroutine 192:
  github.com/pingcap/tidb/sessionctx/variable.glob..func226()
      /go/tidb/sessionctx/variable/sysvar.go:1621 +0xa4
  github.com/pingcap/tidb/sessionctx/variable.(*SysVar).SetSessionFromHook()
      /go/tidb/sessionctx/variable/sysvar.go:199 +0xaf
  github.com/pingcap/tidb/sessionctx/variable.(*SessionVars).SetSystemVar()
      /go/tidb/sessionctx/variable/session.go:1588 +0x59
  github.com/pingcap/tidb/executor.loadVariables()
      /go/tidb/executor/plan_replayer.go:544 +0x70d
  github.com/pingcap/tidb/executor.(*PlanReplayerLoadInfo).Update()
      /go/tidb/executor/plan_replayer.go:618 +0x137
  github.com/pingcap/tidb/server.(*clientConn).handlePlanReplayerLoad()
      /go/tidb/server/conn.go:1742 +0xd7
  github.com/pingcap/tidb/server.(*clientConn).handleQuerySpecial()
      /go/tidb/server/conn.go:2024 +0x6cd
  github.com/pingcap/tidb/server.(*clientConn).handleStmt()
      /go/tidb/server/conn.go:1978 +0x5a8
  github.com/pingcap/tidb/server.(*clientConn).handleQuery()
      /go/tidb/server/conn.go:1819 +0x97c
  github.com/pingcap/tidb/server.(*clientConn).dispatch()
      /go/tidb/server/conn.go:1324 +0x1228
  github.com/pingcap/tidb/server.(*clientConn).Run()
      /go/tidb/server/conn.go:1079 +0x313
  github.com/pingcap/tidb/server.(*Server).onConn()
      /go/tidb/server/server.go:548 +0x14fc
  github.com/pingcap/tidb/server.(*Server).startNetworkListener·dwrap·35()
      /go/tidb/server/server.go:451 +0x47
Previous read at 0x00c0095f6840 by goroutine 132:
  github.com/pingcap/tidb/executor.(*ExecStmt).LogSlowQuery()
      /go/tidb/executor/adapter.go:942 +0x184
  github.com/pingcap/tidb/executor.(*ExecStmt).FinishExecuteStmt()
      /go/tidb/executor/adapter.go:897 +0x5cd
  github.com/pingcap/tidb/session.runStmt()
      /go/tidb/session/session.go:1717 +0x8c8
  github.com/pingcap/tidb/session.(*session).ExecuteStmt()
      /go/tidb/session/session.go:1582 +0xd93
  github.com/pingcap/tidb/session.(*session).ExecuteInternal()
      /go/tidb/session/session.go:1295 +0x4c4
  github.com/pingcap/tidb/ddl/util.RemoveFromGCDeleteRange()
      /go/tidb/ddl/util/util.go:117 +0x163
  github.com/pingcap/tidb/ddl/util.CompleteDeleteRange()
      /go/tidb/ddl/util/util.go:112 +0x1a9
  github.com/pingcap/tidb/ddl.(*delRange).doTask()
      /go/tidb/ddl/delete_range.go:233 +0x4d3
  github.com/pingcap/tidb/ddl.(*delRange).doDelRangeWork()
      /go/tidb/ddl/delete_range.go:186 +0x66e
  github.com/pingcap/tidb/ddl.(*delRange).startEmulator()
      /go/tidb/ddl/delete_range.go:150 +0x1c4
  github.com/pingcap/tidb/ddl.(*delRange).start·dwrap·31()
      /go/tidb/ddl/delete_range.go:126 +0x39 
```

### 1. Minimal reproduce step (Required)

```make race```

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #30345

Problem Summary:

### What is changed and how it works?

make EnableSlowLog atomic to fix data race.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
It is a problem that Atomic.Bool cannot Unmarshal as toml.

**Comment by tangenta** [comment]
The global config should be read-only after TiDB is bootstrapped. Maybe introducing a golang variable can solve this issue.

**Comment by hawkingrei** [comment]
> @hawkingrei The global config should be read-only after TiDB is bootstrapped. Introducing a global variable(Golang) may solve this issue.



> @hawkingrei The global config should be read-only after TiDB is bootstrapped. Introducing a global variable(Golang) may solve this issue.

@tangenta I find that some config will atomic write data.

https://github.com/pingcap/tidb/blob/master/sessionctx/variable/sysvar.go#L1363

if we implement it, we won't think about it later.

**Comment by hawkingrei** [comment]
> @hawkingrei The global config should be read-only after TiDB is bootstrapped. Introducing a global variable(Golang) may solve this issue.

A global variable is a better way to solve it.

**Comment by hawkingrei** [comment]
global config is not read-only.  dynamic config needs to make global config write-read.

File: config/config.go [hunk 1]
```diff
@@ -301,6 +302,42 @@ func (b *nullableBool) UnmarshalJSON(data []byte) error {
 	return err
 }
 
+// AtomicBool is a helper type for atomic operations on a boolean value.
+type AtomicBool struct {
```
Comment (bb7133) [inline config/config.go:306]: Please add some test cases for AtomicBool.

Comment (hawkingrei) [reply config/config.go:306]: Add some tests for ```AtomicBool``` to encode/decode toml and other tests have been implemented.

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**